test/unit/specs/store/search-store.spec.js

```diff
@@ -1,573 +1,576 @@
+            { checked: false, code: 'by-sa', name: 'filters.licenses.by-sa' },
+            { checked: false, code: 'by-nc', name: 'filters.licenses.by-nc' },
+            { checked: false, code: 'by-nd', name: 'filters.licenses.by-nd' },
+            {
+              checked: false,
+              code: 'by-nc-sa',
+              name: 'filters.licenses.by-nc-sa',
+            },
+            {
+              checked: false,
+              code: 'by-nc-nd',
+              name: 'filters.licenses.by-nc-nd',
+            },
+          ],
+          mature: true,
+          searchBy: [
+            {
+              checked: true,
+              code: 'creator',
+              name: 'filters.searchBy.creator',
+            },
+          ],
+          sizes: [
+            { checked: false, code: 'small', name: 'filters.sizes.small' },
+            { checked: false, code: 'medium', name: 'filters.sizes.medium' },
+            { checked: false, code: 'large', name: 'filters.sizes.large' },
+          ],
+        },
+      })
+    })
+
+    it('CLEAR_FILTERS resets filters to initial state', async () => {
+      state.filters.licenses = [
+        { code: 'by', checked: true },
+        { code: 'by-nc', checked: true },
+        { code: 'by-nd', checked: true },
+      ]
+      actions[CLEAR_FILTERS]({
+        commit: commitMock,
+        dispatch: dispatchMock,
+        state,
+      })
+      expect(commitMock).toHaveBeenCalledWith(REPLACE_FILTERS, {
+        newFilterData: filterData,
+      })
+      expect(dispatchMock).toHaveBeenCalledWith(UPDATE_QUERY)
+    })
+
+    it('CLEAR_FILTERS sets providers filters checked to false', async () => {
+      state.filters.imageProviders = [
+        { code: 'met', name: 'Metropolitan', checked: true },
+        { code: 'flickr', name: 'Flickr', checked: false },
+      ]
+
+      actions[CLEAR_FILTERS]({
+        commit: commitMock,
+        dispatch: dispatchMock,
+        state,
+      })
+      const expectedFilters = {
+        ...state.filters,
+        imageProviders: [
+          { code: 'met', name: 'Metropolitan', checked: false },
+          { code: 'flickr', name: 'Flickr', checked: false },
+        ],
+      }
+      expect(commitMock).toHaveBeenCalledWith(REPLACE_FILTERS, {
+        newFilterData: expectedFilters,
+      })
+      expect(dispatchMock).toHaveBeenCalledWith(UPDATE_QUERY)
+    })
+
+    it.each`
+      filterType           | code              | idx
+      ${'licenses'}        | ${'cc0'}          | ${0}
+      ${'licenseTypes'}    | ${'modification'} | ${1}
+      ${'imageExtensions'} | ${'svg'}          | ${3}
+      ${'imageCategories'} | ${'photograph'}   | ${0}
+      ${'searchBy'}        | ${'creator'}      | ${0}
+      ${'mature'}          | ${undefined}      | ${-1}
+      ${'aspectRatios'}    | ${'tall'}         | ${0}
+      ${'sizes'}           | ${'medium'}       | ${1}
+    `(
+      "TOGGLE_FILTER should set filter '$code' of type '$filterType",
```

Comment:
A m a z i n g ✨

Reply:
Thank you, @krysal! I tried using the parametrized before, but it didn't work for some reason. This time, I got really tired of changing the same tests over and over :)